x/dex/types/shares_test.go

```diff
@@ -133,170 +133,67 @@ func TestMaximalSharesFromExactRatioJoin(t *testing.T) {
 	}
 }
 
+// See https://www.notion.so/nibiru/Single-Asset-Join-Math-2075178cb9684062b9b65ad23ff14417
 func TestSwapForSwapAndJoin(t *testing.T) {
 	for _, tc := range []struct {
 		name             string
 		poolAssets       []PoolAsset
 		existingShares   int64
-		tokensIn         sdk.Coins
+		tokenIn          sdk.Coin
+		swapFee          sdk.Dec
 		expectedX0Denom  string
 		expectedX0Amount int64
 		err              error
 	}{
 		{
-			name: "tokens bbb need to be swapped, but no swap necessary",
+			name: "single asset join",
 			poolAssets: []PoolAsset{
 				{
-					Token:  sdk.NewInt64Coin("aaa", 1000),
+					Token:  sdk.NewInt64Coin("atom", 3_256_428),
 					Weight: sdk.NewInt(1 << 30),
 				},
 				{
-					Token:  sdk.NewInt64Coin("bbb", 1000),
+					Token:  sdk.NewInt64Coin("osmo", 31_278_086),
 					Weight: sdk.NewInt(1 << 30),
 				},
 			},
-			existingShares: 100,
-			tokensIn: sdk.NewCoins(
-				sdk.NewInt64Coin("aaa", 100),
-				sdk.NewInt64Coin("bbb", 101),
-			),
-			expectedX0Denom:  "bbb",
-			expectedX0Amount: 0,
-		},
-		{
-			name: "tokens aaa need to be swapped, but no swap necessary",
-			poolAssets: []PoolAsset{
-				{
-					Token:  sdk.NewInt64Coin("aaa", 1000),
-					Weight: sdk.NewInt(1 << 30),
-				},
-				{
-					Token:  sdk.NewInt64Coin("bbb", 1000),
-					Weight: sdk.NewInt(1 << 30),
-				},
-			},
-			existingShares: 100,
-			tokensIn: sdk.NewCoins(
-				sdk.NewInt64Coin("aaa", 101),
```

Comment:
I like the new tests that are added. But, we should keep the current test cases (the "aaa" and "bbb" cases) there and update them to have new outputs so that the effect of this PR will be clear just from the side-by-side comparison.